src/parallel/parallel.cpp

```diff
@@ -186,7 +228,20 @@ ParallelRobinHoodHashTable::distance_zero_insert(KeyType key,
                                                  ValueType value,
                                                  size_t dist_zero_slot)
 {
-    // TODO
+    bool key_exists = false;
+    bool inserted = false;
+    KeyValue blank_kv;
+    KeyValue new_kv = {key, value};
+    if (this->buckets_[dist_zero_slot].key_value.load().key == blank_kv.key) {
+        KeyValue insert_kv = compare_and_set_key_val(dist_zero_slot, blank_kv, new_kv);
+        if (insert_kv.key == 0) {
+            inserted = true;
+        }
+    }
+    if (this->buckets_[dist_zero_slot].key_value.load().key == key) {
+        key_exists = true;
+    }
+    return std::make_pair(inserted, key_exists);
```

Comment:
Instead of a pair, could we create an `enum class` with similar semantics? That way we return a simpler object and have more intuitive semantics (that express intent).